tests/unit/task-property-test.js

```diff
@@ -0,0 +1,194 @@
+import Ember from 'ember';
+import { task, interval } from 'ember-concurrency';
+import { TaskProperty } from 'ember-concurrency/-task-property';
+import { module, test } from 'qunit';
+
+module('Unit: task property', {
+  beforeEach() {
+    let taskRunCounter = 0;
+    this.waiter = () => taskRunCounter === 0;
+
+    Ember.Test.registerWaiter(this.waiter);
+
+    TaskProperty.prototype.withTestWaiter = function() {
+      if (Ember.testing) {
+        let originalTaskFn = this.taskFn;
+
+        this.taskFn = function * (...args) {
+          taskRunCounter += 1;
+          try {
+            return yield * originalTaskFn.apply(this, args);
+          } finally {
+            taskRunCounter -= 1;
+          }
+        };
+      }
+
+      return this;
+    }
+  },
+
+  afterEach() {
+    Ember.Test.unregisterWaiter(this.waiter);
+    delete TaskProperty.prototype.withTestWaiter;
+  }
+});
```

Comment:
I just pushed the changes that switch over to using a counter, but don't delete the concurrency scenario tests. Let me know if you want me to delete the concurrency scenario tests as well.